File: ZeroTier One/NodeStatus.m
/*
 * ZeroTier One - Network Virtualization Everywhere
 * Copyright (C) 2011-2016  ZeroTier, Inc.  https://www.zerotier.com/
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
#import "NodeStatus.h"

@implementation NodeStatus

- (id)initWithJsonData:(NSDictionary*)jsonData
{
    self = [super init];

    if(self) {
        _address = (NSString*)[jsonData objectForKey:@"address"];
        _publicIdentity = (NSString*)[jsonData objectForKey:@"publicIdentity"];
        _online = [(NSNumber*)[jsonData objectForKey:@"online"] boolValue];
        _tcpFallbackActive = [(NSNumber*)[jsonData objectForKey:@"tcpFallbackActive"] boolValue];
        _versionMajor = [(NSNumber*)[jsonData objectForKey:@"versionMajor"] intValue];
        _versionMinor = [(NSNumber*)[jsonData objectForKey:@"versionMinor"] intValue];
        _versionRev = [(NSNumber*)[jsonData objectForKey:@"versionRev"] intValue];
        _version = (NSString*)[jsonData objectForKey:@"version"];
        _clock = [(NSNumber*)[jsonData objectForKey:@"clock"] unsignedLongLongValue];
    }

    return self;
}
@end

File: ZeroTier One/NodeStatus.swift
//
//  NodeStatus.swift
//  ZeroTier One
//
//  Created by Grant Limberg on 6/30/16.
//  Copyright © 2016 ZeroTier, Inc. All rights reserved.
//

import Cocoa

class NodeStatus: NSObject {
    var address: String = ""
    var publicIdentity: String = ""
    var online: Bool = false
    var tcpFallbackActive: Bool = false
    var versionMajor: Int = 0
    var versionMinor: Int = 0
    var versionRev: Int = 0
    var version: String = ""
    var clock: UInt64 = 0

    init(jsonData: [String: AnyObject]) {
        if let a = jsonData["address"] as? String {
            self.address = a
        }

        if let p = jsonData["publicIdentity"] as? String {
            self.publicIdentity = p
        }

        if let o = jsonData["online"] as? NSNumber {
            self.online = o.boolValue
        }

        if let t = jsonData["tcpFallbackActive"] as? NSNumber {
            self.tcpFallbackActive = t.boolValue
        }

        if let v = jsonData["versionMajor"] as? NSNumber {
            self.versionMajor = v.integerValue
        }

        if let v = jsonData["versionMinor"] as? NSNumber {
            self.versionMinor = v.integerValue
        }

        if let v = jsonData["versionRev"] as? NSNumber {
            self.versionRev = v.integerValue
        }

        if let v = jsonData["version"] as? String {
            self.version = v
        }

        if let c = jsonData["version"] as? NSNumber {
            self.clock = c.unsignedLongLongValue
        }
    }
}
